gems/aws-sdk-core/lib/aws-sdk-core/errors.rb

```diff
@@ -208,12 +208,40 @@ class InvalidProcessCredentialsPayload < RuntimeError; end
     # Raised when a client is constructed and region is not specified.
     class MissingRegionError < ArgumentError
       def initialize(*args)
-        msg = 'missing region; use :region option or '\
-              "export region name to ENV['AWS_REGION']"
+        msg = 'No region was provided. Configure the `:region` option or '\
+          "export the region name to ENV['AWS_REGION']"
         super(msg)
       end
     end
 
+    # Raised when a client is contsructed and the region is not valid.
+    class InvalidRegionError < ArgumentError
+      def initialize(*args)
+        super(<<-MSG)
+Invalid `:region` option was provided.
+
+* Not every service is available in every region.
+
+* Never suffix region names with availability zones.
+  Use "us-east-1", not "us-east-1a"
+
+Known AWS regions include (not specific to this service):
+
+#{possible_regions}
+        MSG
+      end
+
+      private
+
+      def possible_regions
```

Comment:
This is duplicated - and seems like it should probably live somewhere other than in individual error classes

Reply:
It is duplicated. I can't really re-use it here (don't want to change how errors are inherited). This logic could maybe live in aws-partitions somewhere if you have any ideas.